server/src/main/java/org/apache/cloudstack/network/router/deployment/RouterDeploymentDefinition.java

```diff
@@ -353,7 +353,8 @@ protected boolean prepareDeployment() {
         setupAccountOwner();
 
         // Check if public network has to be set on VR
-        isPublicNetwork = networkModel.isProviderSupportServiceInNetwork(guestNetwork.getId(), Service.SourceNat, Provider.VirtualRouter);
+        isPublicNetwork = networkModel.isProviderSupportServiceInNetwork(guestNetwork.getId(), Service.SourceNat, Provider.VirtualRouter)
```

Comment:
minor: should we have a method to check multiple services at once?

Reply:
@shwstppr 
I will check

Reply:
done
added a new method `networkModel.isAnyServiceSupportedInNetwork`